backend/pkg/console/schema_registry.go

```diff
@@ -0,0 +1,569 @@
+	Schemas             []*SchemaRegistryVersionedSchema      `json:"schemas"`
+}
+
+const (
+	// SchemaVersionsAll can be specified as version to retrieve all schema versions.
+	SchemaVersionsAll string = "all"
+	// SchemaVersionsLatest can be specified as version to retrieve the latest active schema.
+	SchemaVersionsLatest string = "latest"
+)
+
+// GetSchemaRegistrySubjectDetails retrieves the schema details for the given subject, version tuple.
+// Use version = 'latest' to retrieve the latest schema.
+// Use version = 'all' to retrieve all schemas for this subject.
+// Use version = 3 to retrieve the specific version for the given subject
+func (s *Service) GetSchemaRegistrySubjectDetails(ctx context.Context, subjectName, version string) (*SchemaRegistrySubjectDetails, error) {
+	// 1. Retrieve all schema versions registered for the given subject
+	versions, err := s.getSchemaRegistrySchemaVersions(ctx, subjectName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve subject versions: %w", err)
+	}
+	latestActiveVersion := -1
+	softDeletedVersions := make(map[int]bool)
+	for _, v := range versions {
+		if v.IsSoftDeleted {
+			softDeletedVersions[v.Version] = true
+			continue
+		}
+		if v.Version > latestActiveVersion {
+			latestActiveVersion = v.Version
+		}
+	}
+
+	// 2. Retrieve schemas and compat level concurrently
+	var compatLevel schema.CompatibilityLevel
+
+	grp, grpCtx := errgroup.WithContext(ctx)
+
+	grp.Go(func() error {
+		// 2. Retrieve subject config (subject compatibility level)
+		configRes, err := s.kafkaSvc.SchemaService.GetSubjectConfig(grpCtx, subjectName)
+		if err != nil {
+			s.logger.Warn("failed to get subject config", zap.String("subject", subjectName), zap.Error(err))
+			return nil
+		}
+		compatLevel = configRes.Compatibility
+		return nil
+	})
+
+	// If version 'all' request schemas for all versions individually
+	versionsToRetrieve := []string{version}
+	if version == SchemaVersionsAll {
+		versionsToRetrieve = make([]string, len(versions))
+		for i, v := range versions {
+			versionsToRetrieve[i] = strconv.Itoa(v.Version)
+		}
+	}
+
+	schemasCh := make(chan *SchemaRegistryVersionedSchema, len(versionsToRetrieve))
+	for _, v := range versionsToRetrieve {
+		copiedVersion := v
+		grp.Go(func() error {
+			schemaRes, err := s.GetSchemaRegistrySchema(ctx, subjectName, copiedVersion, true)
+			if err != nil {
+				return fmt.Errorf("failed to retrieve version %q", copiedVersion)
+			}
+			// To avoid making further requests to figure out whether this is a soft-deleted
+			// schema or not we are looking up this piece of information in our existing
+			// versions map.
+			schemaRes.IsSoftDeleted = softDeletedVersions[schemaRes.Version]
+			schemasCh <- schemaRes
+			return nil
+		})
+	}
+
+	if err := grp.Wait(); err != nil {
+		return nil, err
+	}
+	close(schemasCh)
+
+	// Send requests to retrieve schemas
+	schemas := make([]*SchemaRegistryVersionedSchema, 0)
```

Comment:
Another minor perf nit, I think we can make with capacity of `len(versionsToRetrieve)`